### DISCUSSION TITLE
Quality Difference

### DISCUSSION THREAD

**Discussion by codestuff001** [smy20011/dreambooth-gui#44] [General]
I've been testing Dreambooth-GUI the past few days and not getting the best results. I've been training it on selfie photos, and the results are not very accurate to the source images, so I tried testing the same set of images with Shavim's notebook, and the results were night and day better.

I tested one more time with Dreambooth-GUI, this time I even used the same class images generated by Shavim's notebook, and still the quality is noticeably worse.

One thing that's mentioned a lot to improve quality, is enabling `--train_text_encoder`. Does Dreambooth-GUI use text encoder training?

**Comment by futurevessel** [comment]
Not by default, but you can just add it to the parameters box and it will be used. It requires more VRAM though, so depending on your hardware it might not work. And yes, it greatly improves quality.

**Reply by codestuff001** [reply]
I did try enabling training text encoder, but I got this error:
https://github.com/smy20011/dreambooth-gui/issues/41#issue-1436036790

I couldn't tell if it's because it's not supported, or not enough vram. I have a 3080ti with 12GB.

**Reply by futurevessel** [reply]
Yes it supports it, I also have 12gb vram, so when I use --train_text_encoder I also need to enable --gradient_checkpointing else I run of of memory, try pasting the following into your training parameter box:

--mixed_precision=fp16
--train_batch_size=1
--gradient_accumulation_steps=1
--use_8bit_adam
--resolution=512
--gradient_checkpointing
--train_text_encoder

**Comment by bmoggedhwm** [comment]
> I've been testing Dreambooth-GUI the past few days and not getting the best results. I've been training it on selfie photos, and the results are not very accurate to the source images, so I tried testing the same set of images with Shavim's notebook, and the results were night and day better.
> 
> I tested one more time with Dreambooth-GUI, this time I even used the same class images generated by Shavim's notebook, and still the quality is noticeably worse.
> 
> One thing that's mentioned a lot to improve quality, is enabling `--train_text_encoder`. Does Dreambooth-GUI use text encoder training?

It's not just you seeing the same thing adding train_text_encoder does help but it doesn't match the quality of the notebooks I have.

Mind you I can't get it to run locally any other way than with Dreambooth Gui so I am very grateful for that...

**Comment by smy20011** [comment]
You may want to change the learning rate from 1e-6 to 6e-6. I changed it
because there is a version of diffusers code that use 1e-6 for better
quality.

**Reply by futurevessel** [reply]
Wait, I'm not sure I follow, are you saying that on dreambooth-gui, 1e-6 is different from diffusers version ? 

I was under the impression dreambooth-gui was based upon https://github.com/ShivamShrirao) ?

**Reply by smy20011** [reply]
Yeah, there was a version that ShivamShrirao uses 1e-6 for better quality (but you need train longer). I change the learning rate after that.

**Reply by smy20011** [reply]
Yeah, they are using 1e-6 in the codelab
https://github.com/ShivamShrirao/diffusers/blob/main/examples/dreambooth/DreamBooth_Stable_Diffusion.ipynb

**Comment by smy20011** [comment]
If possible, do you mind share the version and config of Collab you are using. And the training command in the gui.

**Reply by bmoggedhwm** [reply]
Sure

https://colab.research.google.com/github/TheLastBen/fast-stable-diffusion/blob/main/fast-DreamBooth.ipynb

I made my app run at 2e-6, seed 96576, tried at 1200, 1800, and 3000. max_train_steps was an option not available nor was doing a lr_scheduler="polynomial". 

Like I said the differences were pretty dramatic. I love the work you have done and your implementation I could not get other versions of dreambooth working on my machine. I am very happy that I could get something to come out.
